pkg/node/node.go

```diff
@@ -253,13 +253,13 @@ func (s *Service) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeR
 		return nil, status.Error(codes.InvalidArgument, "staging target path is required")
 	}
 
-	id, arrayID, protocol, _, _, _ := array.ParseVolumeID(ctx, id, s.DefaultArray(), req.VolumeCapability)
+	id, arrayID, protocol, remoteVolumeID, _, _ := array.ParseVolumeID(ctx, id, s.DefaultArray(), req.VolumeCapability)
```

Comment:
should we check parser error here or is there a reason we skipped it?

Reply:
Full Volume ID was already checked before it was parsed. If there is any issue in the ID, we wouldn't have reached this Node call. Anyways I've added an error check as a precaution.